src/components/Autocomplete/components/ComboBox/ComboBox.tsx

```diff
@@ -123,17 +115,14 @@ export default class ComboBox extends React.PureComponent<Props, State> {
     popoverWasActive: false,
   };
 
-  private subscriptions: {(): void}[] = [];
   private popoverScrollContainer: React.RefObject<
     HTMLDivElement
   > = React.createRef();
 
-  getChildContext(): Context {
+  get getContext(): ComboBoxContext {
```

Comment:
Is there any value in having this be a function if it's only used once? Should we inline it into render():

```js
const context = {
  comboBoxId: this.state.comboBoxId
  selectedOptionId: this.selectedOptionId,
} as ComboBoxContext;

// ...

return (
  <Provider value={context}>
    {/*... */}
  </Provider>
)
```

Reply:
Not really in this situation. Other components may have a more complex context, but `ComboBox` is rather simple. It's more for a common convention / easily searchable. I wouldn't be opposed to changing it if others don't think we get value from it.

Reply:
Looking at your other PRs it looks like their equivalent of this function is only used in their render function too. It seems the simple style of single usage is the common case so I think we should stick with that. The hooks-style of coding seems to be sticking everything into the render function anyway so making it follow that pattern feels nicer for consistency

Reply:
I wouldn't think to search for "getContext" to find contexts - I'd search for React.createContext and follow its usage.

Inlining these can be done as part of the follow up PR so I'm going to resolve this thread for now.